framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -334,13 +331,12 @@ public AnnotatedTypeFactory(BaseTypeChecker checker) {
 
         this.cacheDeclAnnos = new HashMap<Element, Set<AnnotationMirror>>();
 
-        int cacheSize = getCacheSize();
+        this.shouldCache = !checker.hasOption("atfDoNotCache");
+        int cacheSize = shouldCache ? getCacheSize() : 0;
```

Comment:
I would make this more explicit: when shouldcache is false, set fields to `null` otherwise determine cache size and create them.